src/main/java/com/conveyal/datatools/manager/models/FeedSource.java

```diff
@@ -459,108 +459,73 @@ public String latestVersionId() {
         return latest != null ? latest.id : null;
     }
 
+    /**
+     * The deployed feed version.
+     * This cannot be returned because of a circular reference between feed source and feed version. Instead, individual
+     * parameters (version id, start date and end date) are returned.
+     */
+    @JsonIgnore
+    @BsonIgnore
+    private FeedVersionDeployed deployedFeedVersion;
+
+    /**
+     * This value is set to true once an attempt has been made to get the deployed feed version. This prevents subsequent
+     * attempts where the deployed feed version is not available.
+     */
+    @JsonIgnore
+    @BsonIgnore
+    private boolean deployedFeedVersionDefined;
```

Comment:
Why do we do this instead of a null check on deployedFeedVersion?

Reply:
This is to prevent calls to `retrieveDeployedFeedVersion()` where there are no deployed feed versions. Using null resulted in the following scenerio: Call to `getDeployedFeedVersionId()` returned null, followed by a call to `getDeployedFeedVersionStartDate()` returning null and a call to getDeployedFeedVersionEndDate() also returning null.

Reply:
I think this may just be issues with my java understanding how is this field better than those 3 methods returning null? Is it to avoid mongo queries?